Detections/AzureActivity/RareOperations.yaml

```diff
@@ -14,33 +14,32 @@ queryPeriod: 14d
 triggerOperator: gt
 triggerThreshold: 0
 tactics:
-  - CredentialAccess 
+  - CredentialAccess
   - Persistence
 relevantTechniques:
   - T1003
   - T1098
 query: |
- 
+
   let starttime = 14d;
   let endtime = 1d;
   // The number of operations below which an IP address is considered an unusual source of role assignment operations
   let alertOperationThreshold = 5;
-  let SensitiveOperationList = dynamic(
-  ["List keys", "List Storage Account Keys", "Register Subscription", "Create or Update Snapshot", "Create or Update Network Security Group"]);
+  let SensitiveOperationList =  dynamic(["microsoft.compute/snapshots/write", "microsoft.network/networksecuritygroups/write", "microsoft.storage/storageaccounts/listkeys/action"]);
   let SensitiveActivity = AzureActivity
-  | where OperationNameValue in~ (SensitiveOperationList)
+  | where tolower(OperationNameValue) in (SensitiveOperationList) or tolower(OperationNameValue) endswith "listkeys/action"
```

Comment:
use in~ instead of tolower().  After the "or" do not use tolower() as endswith is not case sensitive and also recommend you use hassuffix instead of endswith according to performance docs.

Reply:
done